js/snake.js

```diff
@@ -1,305 +1,417 @@
      * @returns {int} - pixel value of beginning of the grid-field
      */
-    gridToRaw : function  (grid){
-    return grid * this.stepWidth;
-    },
+    gridToRaw(grid) {
+        return grid * this.stepWidth;
+    };
 
     /**
-     * @description - Converts a grid-object to the pixel coords of its left upper corner
-     * @param {{x: int, y: int}} grid 
-     * @returns Coord - grid-coord object
+     * Converts a grid-object to the pixel coords of its left upper corner
+     * @param {Coord} grid 
+     * @returns {Coord} - grid-coord object
      */
-    gridToRawXY : function(grid){
-    return new Coord(parseInt(this.gridToRaw(grid.x)), parseInt(this.gridToRaw(grid.y)));
-    },
+    gridToRawXY(grid) {
+        return new Coord(parseInt(this.gridToRaw(grid.x)), parseInt(this.gridToRaw(grid.y)));
+    };
 
-    renderSegment : function  (segment, color=DBIWHITE){
-        point = this.gridToRawXY(segment);
+    renderSegment(segment, color = "magenta") {
+        let point = this.gridToRawXY(segment);
         //console.log(point);
         var dim = this.stepWidth;
         this.context.fillStyle = color;
-        this.context.fillRect(point.x, point.y,dim, dim);
-    }
-};
+        this.context.fillRect(point.x, point.y, dim, dim);
+    };
 
-class Snake {
+}
+
+class Snake{    
     body = [];
-    heading = new Coord(0,1);
-    food;
-    initLength = 3;
-    constructor(length=3){
-        this.initLength = length;
-        let y = Math.floor(GRIDSIZE/2)-length;
-        for(var i=0; i<length; i++){
-            this.body.unshift(new Coord(Math.floor(GRIDSIZE/2), mod(y+i,GRIDSIZE)));
+    initLength;
+    heading = new Coord(0, 1);
+    gridSize;
+    color;
+    headColor;
+    constructor(initLength = 3, gridSize, color="white", headColor="grey") {
+        this.initLength = initLength;
+        this.gridSize = gridSize;
+        this.color = color;
+        this.headColor = headColor;
+        let y = Math.floor(this.gridSize / 2) - this.initLength;
+        for (var i = 0; i < this.initLength; i++) {
+            this.body.unshift(new Coord(Math.floor(this.gridSize / 2), SnakeUtility.mod(y + i, this.gridSize)));
         }
-        this.placeFood();
     }
 
     /**
-     * @description Schlange auf Canvas zeichnen
-     * @param {*} field Spielfeld
-     * @param {*} color Farbe für Schwanz
-     * @param {*} headcolor Farbe für Kopf
-     */
-    renderBody(field, color=DBIWHITE, headcolor="green"){
-        var head = this.body[0];
-        var tail = this.body.slice(1);
-        tail.forEach(segment => {
-            field.renderSegment(segment,color);
-        });
-        field.renderSegment(head,headcolor);
-    }    
-    
-    /**
-    * @description Schlangenbewegung
-    * @param {char} dir : r,l -> lenken
+    * moves the snake
     */
-   move(dir){
-        field.key = "no";
-        if (dir == "l"){
-           this.heading.setXY(this.heading.y, -this.heading.x);
+    move(dir) {
+        if (dir == "l") {
```

Comment:
Kann man so machen. Wenn du weißt, dass immer ein String kommt (also in Dir) kannst du auch einen typsicheren Vergleich fahren (===).

Reply:
Kommt der klar damit, wenn ein "undefined" kommt?

Reply:
Sollte er, ja.